src/error.rs

```diff
@@ -4,23 +4,17 @@ pub type CompactSymbolResult<T> = Result<T, CompactSymbolError>;
 
 #[derive(Fail, Debug)]
 pub enum CompactSymbolError {
-    #[fail(display = "Strings do not match")]
-    UnmatchedString,
+    #[fail(display = "Unmatched strings: Expected {}, but received {}", _0, _1)]
+    UnmatchedString(String, String),
```

Comment:
Here I added two parameters to the `UnmatchedString` error. Where the first param is the expected value and second is actual. Hope this pins down where the errors are occurring.